contracts/dependency-registry/DependencyRegistryV1.sol

```diff
@@ -0,0 +1,1092 @@
+        );
+
+        bytes32 tokenHash;
+        try
+            IDependencyRegistryCompatibleV0(_contractAddress).tokenIdToHash(
+                _tokenId
+            )
+        returns (bytes32 _tokenHash) {
+            tokenHash = _tokenHash;
+        } catch {
+            // Noop try again for older contracts.
+        }
+
+        if (tokenHash == bytes32(0)) {
+            try
+                IGenArt721CoreTokenHashProviderV1(_contractAddress)
+                    .tokenIdToHash(_tokenId)
+            returns (bytes32 tokenHash) {
+                tokenHash = tokenHash;
+            } catch {
+                // Noop try again for older contracts.
+            }
+        }
+
+        if (tokenHash == bytes32(0)) {
+            try
+                IGenArt721CoreTokenHashProviderV0(_contractAddress)
+                    .showTokenHashes(_tokenId)
+            returns (bytes32[] memory tokenHashes) {
+                tokenHash = tokenHashes[0];
+            } catch {
+                revert("Unable to retrieve token hash.");
+            }
+        }
+
+        WrappedScriptRequest[] memory requests = new WrappedScriptRequest[](5);
+
+        requests[0]
+            .scriptContent = 'let css="html{height:100%}body{min-height:100%;margin:0;padding:0}canvas{padding:0;margin:auto;display:block;position:absolute;top:0;bottom:0;left:0;right:0}",head=document.head,style=document.createElement("style");head.appendChild(style),style.type="text/css",style.appendChild(document.createTextNode(css));';
+
+        requests[1].scriptContent = abi.encodePacked(
+            'let tokenData = {"tokenId":"',
+            Strings.toString(_tokenId),
+            '"',
+            ',"hash":"',
+            Strings.toHexString(uint256(tokenHash)),
+            '"}'
+        );
+
+        bytes memory dependencyScript = this.getDependencyScript(
+            dependencyType
+        );
+        requests[2].scriptContent = dependencyScript;
+        requests[2].wrapType = 2; // <script type="text/javascript+gzip" src="data:text/javascript;base64,[script]"></script>
+
+        bytes memory gunzipScript = IContractScript(ethFsAddress).getScript(
+            "gunzipScripts-0.0.1.js",
+            ""
+        );
+        requests[3].wrapType = 1; // <script src="data:text/javascript;base64,[script]"></script>
+        requests[3].scriptContent = gunzipScript;
+
+        bytes memory projectScript = this.getProjectScript(
+            _contractAddress,
+            projectId
+        );
+        requests[4].scriptContent = projectScript;
+        requests[4].wrapType = 0; // <script>[script]</script>
+
+        IScriptyBuilder scriptyBuilder = IScriptyBuilder(scriptyBuilderAddress);
+
+        uint256 bufferSize = requests[0].scriptContent.length +
+            requests[1].scriptContent.length +
+            requests[2].scriptContent.length +
+            requests[3].scriptContent.length +
+            requests[4].scriptContent.length +
+            500;
```

Comment:
@0xdude I was getting `DynamicBuffer: Appending out of bounds.` reverts without adding this arbitrary 500 bytes to the buffer, anything I'm doing obviously wrong here?

Reply:
Sorry for misleading previous comment. I believe, since all requests use `scriptContent` you can use this now:
`getBufferSizeForHTMLWrapped(requests)`

Scripty won't do contract call and just calculate all the script length and tag sizes. It should match but let me know if you have any problem.